packages/cli/src/config/trustedFolders.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { homedir, platform } from 'os';
+import { getErrorMessage } from '@google/gemini-cli-core';
+import stripJsonComments from 'strip-json-comments';
+import { SettingScope } from './settings.js';
+
+export const TRUSTED_FOLDERS_FILENAME = 'trustedFolders.json';
+export const SETTINGS_DIRECTORY_NAME = '.gemini';
+export const USER_SETTINGS_DIR = path.join(homedir(), SETTINGS_DIRECTORY_NAME);
+export const USER_TRUSTED_FOLDERS_PATH = path.join(
+  USER_SETTINGS_DIR,
+  TRUSTED_FOLDERS_FILENAME,
+);
+
+export function getSystemTrustedFoldersPath(): string {
+  if (process.env.GEMINI_CLI_SYSTEM_TRUSTED_FOLDERS_PATH) {
+    return process.env.GEMINI_CLI_SYSTEM_TRUSTED_FOLDERS_PATH;
+  }
+  if (platform() === 'darwin') {
+    return `/Library/Application Support/GeminiCli/${TRUSTED_FOLDERS_FILENAME}`;
+  } else if (platform() === 'win32') {
+    return `C:/ProgramData/gemini-cli/${TRUSTED_FOLDERS_FILENAME}`;
+  } else {
+    return `/etc/gemini-cli/${TRUSTED_FOLDERS_FILENAME}`;
+  }
+}
+
+export enum TrustLevel {
+  TRUST_FOLDER = 'TRUST_FOLDER',
+  TRUST_PARENT = 'TRUST_PARENT',
+  DO_NOT_TRUST = 'DO_NOT_TRUST',
+}
+
+export interface TrustRule {
+  path: string;
+  trustLevel: TrustLevel;
+}
+
+export interface TrustedFoldersError {
+  message: string;
+  path: string;
+}
+
+export interface TrustedFoldersFile {
+  config: Record<string, TrustLevel>;
+  path: string;
+}
+
+export class LoadedTrustedFolders {
+  constructor(
+    public system: TrustedFoldersFile,
+    public user: TrustedFoldersFile,
+    public errors: TrustedFoldersError[],
+  ) {}
+
+  get rules(): TrustRule[] {
+    const mergedConfig = { ...this.user.config, ...this.system.config };
+    return Object.entries(mergedConfig).map(([path, trustLevel]) => ({
+      path,
+      trustLevel,
+    }));
+  }
+
+  forScope(scope: SettingScope): TrustedFoldersFile {
```

Comment:
why do we need System scope as well as user scope for this?

Reply:
Removed system scope as discussed offline, since Vscode doesn't seem to have something similar.